core/src/main/java/com/yahoo/oak/BlocksPool.java

```diff
@@ -17,7 +17,7 @@
  */
 final class BlocksPool implements BlocksProvider, Closeable {
 
-    private static BlocksPool instance = null;
+    private static volatile BlocksPool instance = null;
```

Comment:
What is the purpose of the change? BlocksPool should be written only once and single threaded. After accesses can be from caches...

Reply:
Actually, it was IntelliJ suggestion that I accepted. But you are right. I will revert this change.